---
title: "separating_predictions_by_sample"
author: "A.J Wing"
date: "2022-12-21"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(data.table)
library(tidyr)
library(ggplot2)
library(tidyverse)
library(jcolors)
library(viridis)
library(cowplot)
library(gridExtra)
```

#import metadata
```{r}
metadata <- read_tsv("~/Desktop/erie_story_r_work/2014_story_metadata_updated_12_15_21.tsv")
names(metadata)[names(metadata) == 'samples'] <- "sample"
metadata$sample <- gsub("Sample_","",metadata$sample)
```

#import all predictions
#august bloom
```{r}
# load data
HABs2014 <- read_tsv("~/Desktop/erie_story_r_work/12_14_22_Aug4_predictions.tsv")
HABs2014_temp <- separate(HABs2014, col=pairs, c("phage", "host"), sep=":concatenated_", remove = FALSE)
HABs2014_aug_bloom <- HABs2014_temp[, c("phage", "host", "Homology", "Predictions", "GCdiff", "k3dist", "k6dist", "InfProbabilities")]
HABs2014_aug_bloom$phage <- gsub(".fasta", "", HABs2014$phage) 
#HABs2014_aug_bloom$phage <- gsub("concatenated_", "", HABs2014$phage)
head(HABs2014_aug_bloom)
```

#load in microcystis phage predictions
```{r}
# load data
ma_phage <- read_tsv("~/Desktop/erie_story_r_work/Aug4_septvirus.tsv")
ma_phage_temp <- separate(ma_phage, col=pairs, c("phage", "host"), sep=":concatenated_", remove = FALSE)
ma_phage_aug_bloom <- ma_phage_temp[, c("phage", "host", "Homology", "Predictions", "GCdiff", "k3dist", "k6dist", "InfProbabilities")]
ma_phage_aug_bloom$phage <- gsub(".fasta", "", ma_phage$phage) 
#HABs2014_aug_bloom$phage <- gsub("concatenated_", "", ma_phage$phage)
head(ma_phage_aug_bloom)
```

#now merge other predictions with microcystis phage
```{r}
HABs2014_aug_bloom <- rbind(HABs2014_aug_bloom,ma_phage_aug_bloom,by="phage")
dim(HABs2014_aug_bloom)
```


#now modify this to include the sample number for each phage
```{r}

HABs2014_aug_bloom$phage_copy <- HABs2014_aug_bloom$phage
HABs2014_aug_bloom$phage_copy <- gsub("concatenated_", "", HABs2014_aug_bloom$phage_copy)
HABs2014_aug_bloom$phage_copy <- gsub("\\-.*", "", HABs2014_aug_bloom$phage_copy)
HABs2014_aug_bloom$phage_copy <- gsub("\\_.*", "", HABs2014_aug_bloom$phage_copy)
HABs2014_aug_bloom$phage_copy <- gsub("september", "49628", HABs2014_aug_bloom$phage_copy)
names(HABs2014_aug_bloom)[names(HABs2014_aug_bloom) == 'phage_copy'] <- "sample"
HABs2014_aug_bloom$phage <- gsub(".fasta", "", HABs2014_aug_bloom$phage)
HABs2014_aug_bloom$host <- gsub(".fasta", "", HABs2014_aug_bloom$host)
```

#now import viral sequence lengths so we can keep only those >10kb
```{r}
aug_viral_seq_lengths <- read_tsv("~/Desktop/erie_story_r_work/AUG4_virus_sequence_lengths.tsv")
names(aug_viral_seq_lengths)[names(aug_viral_seq_lengths) == 'bin'] <- "phage"
```

#and join the imported lengths with the predictions
```{r}
HABs2014_aug_bloom <- left_join(HABs2014_aug_bloom,aug_viral_seq_lengths,by="phage")
dim(HABs2014_aug_bloom)
```

#and filter by 10kb
```{r}
HABs2014_aug_bloom_10kb <- HABs2014_aug_bloom[HABs2014_aug_bloom$length>=10000,]
```

#apply threshold to reduce datasize and boost confidence in predictions
```{r}
thresh <- 0.93

HABs2014_aug_bloom_w_thresh <- HABs2014_aug_bloom_10kb[which(HABs2014_aug_bloom_10kb$InfProbabilities > thresh), ]
print(dim(HABs2014_aug_bloom_w_thresh))
```

#write whole prediction tsv for august
```{r}
write_tsv(HABs2014_aug_bloom_w_thresh, file="~/Desktop/erie_story_r_work/10kb_cutoff_aug_bloom_with_ma_phage_093thresh_1_30_23.tsv", col_names = T)
```

#join with metadata
```{r}
HABs2014_aug_bloom_w_thresh_and_metadata <- left_join(HABs2014_aug_bloom_w_thresh,metadata,by="sample")
dim(HABs2014_aug_bloom)
```

#write whole predictions with metadata included
```{r}
write_tsv(HABs2014_aug_bloom_w_thresh_and_metadata, file="~/Desktop/erie_story_r_work/aug_bloom_with_ma_phage_093thresh_and_metadata_1_9_23.tsv", col_names = T)
```

#now lets break these predictions up by sample
```{r}
aug_49614_093_predictions <- HABs2014_aug_bloom_w_thresh[HABs2014_aug_bloom_w_thresh$sample=="49614",]
aug_49615_093_predictions <- HABs2014_aug_bloom_w_thresh[HABs2014_aug_bloom_w_thresh$sample=="49615",]
aug_49616_093_predictions <- HABs2014_aug_bloom_w_thresh[HABs2014_aug_bloom_w_thresh$sample=="49616",]
aug_49617_093_predictions <- HABs2014_aug_bloom_w_thresh[HABs2014_aug_bloom_w_thresh$sample=="49617",]
aug_49618_093_predictions <- HABs2014_aug_bloom_w_thresh[HABs2014_aug_bloom_w_thresh$sample=="49618",]
aug_49619_093_predictions <- HABs2014_aug_bloom_w_thresh[HABs2014_aug_bloom_w_thresh$sample=="49619",]
aug_49620_093_predictions <- HABs2014_aug_bloom_w_thresh[HABs2014_aug_bloom_w_thresh$sample=="49620",]
aug_49621_093_predictions <- HABs2014_aug_bloom_w_thresh[HABs2014_aug_bloom_w_thresh$sample=="49621",]
aug_49622_093_predictions <- HABs2014_aug_bloom_w_thresh[HABs2014_aug_bloom_w_thresh$sample=="49622",]
aug_49623_093_predictions <- HABs2014_aug_bloom_w_thresh[HABs2014_aug_bloom_w_thresh$sample=="49623",]
aug_49640_093_predictions <- HABs2014_aug_bloom_w_thresh[HABs2014_aug_bloom_w_thresh$sample=="49640",]
```


#write separate tsvs for each august samples predictions
```{r}
write_tsv(aug_49614_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/aug_49614_093_predictions.tsv", col_names = T)
write_tsv(aug_49615_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/aug_49615_093_predictions.tsv", col_names = T)
write_tsv(aug_49616_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/aug_49616_093_predictions.tsv", col_names = T)
write_tsv(aug_49617_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/aug_49617_093_predictions.tsv", col_names = T)
write_tsv(aug_49618_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/aug_49618_093_predictions.tsv", col_names = T)
write_tsv(aug_49619_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/aug_49619_093_predictions.tsv", col_names = T)
write_tsv(aug_49620_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/aug_49620_093_predictions.tsv", col_names = T)
write_tsv(aug_49621_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/aug_49621_093_predictions.tsv", col_names = T)
write_tsv(aug_49622_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/aug_49622_093_predictions.tsv", col_names = T)
write_tsv(aug_49623_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/aug_49623_093_predictions.tsv", col_names = T)
write_tsv(aug_49640_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/aug_49640_093_predictions.tsv", col_names = T)
```




#september bloom
```{r}
# load data
HABs2014 <- read_tsv("~/Desktop/erie_story_r_work/12_14_22_Sep29_predictions.tsv")
HABs2014_temp <- separate(HABs2014, col=pairs, c("phage", "host"), sep=":concatenated_", remove = FALSE)
HABs2014_sep_bloom <- HABs2014_temp[, c("phage", "host", "Homology", "Predictions", "GCdiff", "k3dist", "k6dist", "InfProbabilities")]
HABs2014_sep_bloom$phage <- gsub(".fasta", "", HABs2014$phage) 
#HABs2014_sep_bloom$phage <- gsub("concatenated_", "", HABs2014$phage)
head(HABs2014_sep_bloom)
```

#load in microcystis phage predictions
```{r}
# load data
ma_phage <- read_tsv("~/Desktop/erie_story_r_work/Sep29_septvirus.tsv")
ma_phage_temp <- separate(ma_phage, col=pairs, c("phage", "host"), sep=":concatenated_", remove = FALSE)
ma_phage_sep_bloom1 <- ma_phage_temp[, c("phage", "host", "Homology", "Predictions", "GCdiff", "k3dist", "k6dist", "InfProbabilities")]
ma_phage_sep_bloom1$phage <- gsub(".fasta", "", ma_phage$phage) 
#HABs2014_aug_bloom$phage <- gsub("concatenated_", "", ma_phage$phage)
head(ma_phage_sep_bloom1)
```

```{r}
# load data
ma_phage <- read_tsv("~/Desktop/erie_story_r_work/Sep29_augvirus.tsv")
ma_phage_temp <- separate(ma_phage, col=pairs, c("phage", "host"), sep=":concatenated_", remove = FALSE)
ma_phage_sep_bloom2 <- ma_phage_temp[, c("phage", "host", "Homology", "Predictions", "GCdiff", "k3dist", "k6dist", "InfProbabilities")]
ma_phage_sep_bloom2$phage <- gsub(".fasta", "", ma_phage$phage) 
#HABs2014_aug_bloom$phage <- gsub("concatenated_", "", ma_phage$phage)
head(ma_phage_sep_bloom2)
```

#now merge other predictions with microcystis phage
```{r}
HABs2014_sep_bloom <- rbind(HABs2014_sep_bloom,ma_phage_sep_bloom1,by="phage")
dim(HABs2014_sep_bloom)
```

```{r}
HABs2014_sep_bloom <- rbind(HABs2014_sep_bloom,ma_phage_sep_bloom2,by="phage")
dim(HABs2014_sep_bloom)
```

#now modify to separate predictions by each samples phages
```{r}

HABs2014_sep_bloom$phage_copy <- HABs2014_sep_bloom$phage
HABs2014_sep_bloom$phage_copy <- gsub("concatenated_", "", HABs2014_sep_bloom$phage_copy)
HABs2014_sep_bloom$phage_copy <- gsub("\\-.*", "", HABs2014_sep_bloom$phage_copy)
HABs2014_sep_bloom$phage_copy <- gsub("\\_.*", "", HABs2014_sep_bloom$phage_copy)
HABs2014_sep_bloom$phage_copy <- gsub("september", "49628", HABs2014_sep_bloom$phage_copy)
names(HABs2014_sep_bloom)[names(HABs2014_sep_bloom) == 'phage_copy'] <- "sample"
HABs2014_sep_bloom$phage <- gsub(".fasta", "", HABs2014_sep_bloom$phage)
HABs2014_sep_bloom$host <- gsub(".fasta", "", HABs2014_sep_bloom$host)
```

#now import viral sequence lengths so we can keep only those >10kb
```{r}
sep_viral_seq_lengths <- read_tsv("~/Desktop/erie_story_r_work/SEP29_virus_sequence_lengths.tsv")
names(sep_viral_seq_lengths)[names(sep_viral_seq_lengths) == 'bin'] <- "phage"
```

#and join the imported lengths with the predictions
```{r}
HABs2014_sep_bloom <- left_join(HABs2014_sep_bloom,sep_viral_seq_lengths,by="phage")
dim(HABs2014_sep_bloom)
```

#and filter by 10kb
```{r}
HABs2014_sep_bloom_10kb <- HABs2014_sep_bloom[HABs2014_sep_bloom$length>=10000,]
```

#now apply threshold to reduce network size while increasing confidence in predictions
```{r}
thresh <- 0.93

HABs2014_sep_bloom_w_thresh <- HABs2014_sep_bloom_10kb[which(HABs2014_sep_bloom_10kb$InfProbabilities > thresh), ]
print(dim(HABs2014_sep_bloom_w_thresh))
```

#write whole prediction tsv for september
```{r}
write_tsv(HABs2014_sep_bloom_w_thresh, file="~/Desktop/erie_story_r_work/10kb_cutoff_sep_bloom_with_ma_phage_093thresh_1_30_23.tsv", col_names = T)
```

#join with metadata
```{r}
HABs2014_sep_bloom_w_thresh_w_metadata <- left_join(HABs2014_sep_bloom_w_thresh,metadata,by="sample")
dim(HABs2014_sep_bloom)
```

#write whole prediction tsv for september with metadata
```{r}
write_tsv(HABs2014_sep_bloom_w_thresh_w_metadata, file="~/Desktop/erie_story_r_work/sep_bloom_with_ma_phage_093thresh_w_metadata_1_9_23.tsv", col_names = T)
```

#now lets break these predictions up by sample
```{r}
sep_49624_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49624",]
sep_49625_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49625",]
sep_49626_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49626",]
sep_49627_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49627",]
sep_49628_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49628",]
sep_49629_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49629",]
sep_49630_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49630",]
sep_49631_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49631",]
sep_49632_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49632",]
sep_49633_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49633",]
sep_49634_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49634",]
sep_49635_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49635",]
sep_49641_093_predictions <- HABs2014_sep_bloom_w_thresh[HABs2014_sep_bloom_w_thresh$sample=="49641",]
```


#write separate tsvs for each august samples predictions
```{r}
write_tsv(sep_49624_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49624_093_predictions.tsv", col_names = T)
write_tsv(sep_49625_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49625_093_predictions.tsv", col_names = T)
write_tsv(sep_49626_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49626_093_predictions.tsv", col_names = T)
write_tsv(sep_49627_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49627_093_predictions.tsv", col_names = T)
write_tsv(sep_49628_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49628_093_predictions.tsv", col_names = T)
write_tsv(sep_49629_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49629_093_predictions.tsv", col_names = T)
write_tsv(sep_49630_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49630_093_predictions.tsv", col_names = T)
write_tsv(sep_49631_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49631_093_predictions.tsv", col_names = T)
write_tsv(sep_49632_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49632_093_predictions.tsv", col_names = T)
write_tsv(sep_49633_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49633_093_predictions.tsv", col_names = T)
write_tsv(sep_49634_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49634_093_predictions.tsv", col_names = T)
write_tsv(sep_49635_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49623_093_predictions.tsv", col_names = T)
write_tsv(sep_49641_093_predictions, file="~/Desktop/erie_story_r_work/predictions_by_sample/sep_49640_093_predictions.tsv", col_names = T)
```


#write these to tsv with new columns for evc analysis
```{r}
write_tsv(HABs2014_sep_bloom_fractions_of_interest_w_thresh, file="~/Desktop/erie_story_r_work/sep29_093thresh_predictions_viral_3_100_ajw_12_19_22.tsv", col_names = T)
```

##FOR AUGUST
#import newly written table so we can narrow down to only phages predicted to infect microcystis
```{r}
aug4_093_predictions <- read_tsv("~/Desktop/erie_story_r_work/3kb_cutoff_aug_bloom_with_ma_phage_093thresh_1_27_23.tsv")
```

#filter phages to only predicted microcystis phages
```{r}
aug4_093_microcystis_phages <- subset(aug4_093_predictions, grepl("contigs", aug4_093_predictions$host))
```

#now lets filter the phages again so we get microcystis hosts and non-microcystis hosts
```{r}
aug4_093_ma_phages_all_hosts <- subset(aug4_093_predictions, phage %in% aug4_093_microcystis_phages$phage)
```

#and write this new collection to a tsv for network generation
```{r}
write_tsv(aug4_093_ma_phages_all_hosts, file="~/Desktop/erie_story_r_work/3kb_cutoff_aug4_093_ma_phages_only_1_27_23.tsv", col_names = T)
```



##FOR SEPTEMBER
#import newly written table so we can narrow down to only phages predicted to infect microcystis
```{r}
sep29_093_predictions <- read_tsv("~/Desktop/erie_story_r_work/3kb_cutoff_sep_bloom_with_ma_phage_093thresh_1_27_23.tsv")
```

#filter phages to only predicted microcystis phages
```{r}
sep29_093_microcystis_phages <- subset(sep29_093_predictions, grepl("contigs", sep29_093_predictions$host))
```

#now lets filter the phages again so we get microcystis hosts and non-microcystis hosts
```{r}
sep29_093_ma_phages_all_hosts <- subset(sep29_093_predictions, phage %in% sep29_093_microcystis_phages$phage)
```

#and write this new collection to a tsv for network generation
```{r}
write_tsv(sep29_093_ma_phages_all_hosts, file="~/Desktop/erie_story_r_work/3kb_cutoff_sep29_093_ma_phages_only_1_27_23.tsv", col_names = T)
```